plugins/modules/zammad_ticket.py

```diff
@@ -0,0 +1,318 @@
+#!/usr/bin/python
+
+from __future__ import (absolute_import, division, print_function)
+__metaclass__ = type
+
+DOCUMENTATION = r'''
+---
+author:
+- Scaleup Technologies
+- Melvin Ziemann (@cloucs)
+
+module: zammad_ticket
+
+short_description: Create, update, close a Zammad ticket via the API
+
+version_added: "1.0.0"
+
+description: |
+  This module creates, updates, or closes a Zammad ticket using the API.
+  User credentials and ticket details are passed as parameters. This module can handle
+  ticket creation and updates based on the specified state. It also allows for the
+  closure of existing tickets.
+
+options:
+  fqdn:
+    description: The fully qualified domain name of the Zammad instance.
+    required: true
+    type: str
+  endpoint:
+    description: The API endpoint for the ticket management in Zammad.
+    required: true
+    type: str
+  api_user:
+    description: The username for authenticating with the Zammad API.
+    required: true
+    type: str
+  api_secret:
+    description: The password or API key for authenticating with the Zammad API.
+    required: true
+    type: str
+  state:
+    description: The desired state of the ticket. Use 'present' to create or update a ticket, and 'absent' to close a ticket.
+    required: true
+    type: str
+    choices: ["present", "absent"]
+  ticket_id:
+    description: The unique identifier of the ticket to update or close. Required if state is 'absent' or when updating a ticket.
+    required: false
+    type: str
+  customer:
+    description: The email address of the customer for the ticket.
+    required: true
+    type: str
+  title:
+    description: The title of the ticket.
+    required: true
+    type: str
+  group:
+    description: The group handling the ticket.
+    required: true
+    type: str
+  subject:
+    description: The subject for the ticket's article.
+    required: true
+    type: str
+  body:
+    description: The body content for the ticket's article.
+    required: true
+    type: str
+  internal:
+    description: Indicates whether the article is internal. Defaults to false.
+    required: false
+    type: bool
+    default: false
+  ticket_state:
+    description: The state of the ticket (e.g., open, pending, etc.).
+    required: true
+    type: str
+  priority:
+    description: The priority of the ticket (e.g., low, normal, high).
+    required: true
+	type: str
+
+examples:
+  - name: Create a new ticket
+    zammad_ticket:
+      fqdn: "https://zammad.example.com"
+      endpoint: "/api/v1/tickets"
```

Comment:
I recon /tickets won't change in a versioned API (v1), perhaps we can have a single attribute `api_url: https://zammad.example.com/api`